src/modelGeneration/massBalance/checkBalance.m

```diff
@@ -35,15 +35,23 @@
 if ~exist('fileName','var')
     fileName='';
 end
-
 [nMet,nRxn]=size(model.S);
+if ~exist('missingFormulaeBool','var')
+    missingFormulaeBool=false(nMet,1);
+end
 
-missingFormulaeBool=false(nMet,1);
+for m=1:nMet
+    %cannot handle metabolites with non-standard formulae like
+    %(Gal)3(Glc)1(GlcNAc)2(LFuc)1(Cer)1
+    if ~missingFormulaeBool(m,1) || strcmp(model.metFormulas{m}(1),'(')
```

Comment:
This line will set everything to having a missing formula by default. And later steps test on this property, ignoring actual formulas (see below).
I don't really understand, why any entry indicated not to have a missingFormulae should be set to having one ( probably just the ~ has to be removed here).